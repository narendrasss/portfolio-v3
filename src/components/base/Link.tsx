import styled from 'styled-components';
import { Link as BaseLink } from 'gatsby';
import { Text, TextProps } from './Text';

export interface LinkProps extends TextProps {
  to: string;
}

const Link = styled(Text).attrs((props: LinkProps) => ({
  as: BaseLink,
  to: props.to,
  color: props.color ? props.color : 'black',
}))<LinkProps>`
  text-decoration: none;
`;

export default Link;
