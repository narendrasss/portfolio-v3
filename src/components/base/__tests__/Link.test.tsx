import React from 'react';
import { render } from 'react-testing-library';
import Link from '../Link';

describe('Link', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Link to="localhost" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
