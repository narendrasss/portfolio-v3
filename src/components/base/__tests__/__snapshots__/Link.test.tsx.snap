// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders without crashing 1`] = `
<DocumentFragment>
  <a
    class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
    color="black"
    font-family="sans"
    font-size="0,1"
    href="localhost"
  />
</DocumentFragment>
`;
